src/ethereum/ethNetwork.js

```diff
@@ -534,73 +711,213 @@ export class EthereumNetwork {
     }
 
     const allTransactions = []
-    try {
-      while (1) {
-        const offset = NUM_TRANSACTIONS_TO_QUERY
-        const jsonObj = await this.multicastServers('getTransactions', {
-          currencyCode,
-          address,
-          startBlock,
-          page,
-          offset,
-          contractAddress
-        })
-        const valid = validateObject(jsonObj, schema)
-        if (valid) {
-          const transactions = jsonObj.result
-          for (let i = 0; i < transactions.length; i++) {
-            const tx = this.ethEngine.processEtherscanTransaction(
-              transactions[i],
-              currencyCode
-            )
-            allTransactions.push(tx)
-          }
-          if (transactions.length < NUM_TRANSACTIONS_TO_QUERY) {
-            break
+    let server
+    while (1) {
+      const offset = NUM_TRANSACTIONS_TO_QUERY
+      const response = await this.multicastServers('getTransactions', {
+        currencyCode,
+        address,
+        startBlock,
+        page,
+        offset,
+        contractAddress
+      })
+      server = response.server
+      const jsonObj = response.result
+      const valid = validateObject(jsonObj, schema)
+      if (valid) {
+        const transactions = jsonObj.result
+        for (let i = 0; i < transactions.length; i++) {
+          const tx = this.processEtherscanTransaction(
+            transactions[i],
+            currencyCode
+          )
+          allTransactions.push(tx)
+        }
+        if (transactions.length < NUM_TRANSACTIONS_TO_QUERY) {
+          break
+        }
+        page++
+      } else {
+        throw new Error(
+          `checkTxsEthscan invalid JSON data:${JSON.stringify(jsonObj)}`
+        )
+      }
+    }
+
+    const edgeTransactionsBlockHeightTuple: EdgeTransactionsBlockHeightTuple = {
+      blockHeight: startBlock,
+      edgeTransactions: allTransactions
+    }
+    return {
+      tokenTxs: { [currencyCode]: edgeTransactionsBlockHeightTuple },
+      server
+    }
+  }
+
+  getTokenSymbol(tokenTransfer: AlethioTokenTransfer) {
+    if (tokenTransfer.type === 'EtherTransfer') {
+      return 'ETH'
+    } else {
+      return tokenTransfer.attributes.symbol
+    }
+  }
+
+  isTokenEnabled(tokenTransfer: AlethioTokenTransfer) {
```

Comment:
This should use only contract address from tokenTransfer to determine token.  Then use symbol from getTokenInfo